decompiler/pipeline/dataflowanalysis/dead_loop_elimination.py

```diff
@@ -87,7 +92,7 @@ def _evaluate_branch_condition(
         Evaluate Branch.condition with z3 and return (satisfiable edge, unsatisfiable edge) if one edge is unsatisfiable.
         """
         try:
-            condition = self._logic_converter.convert(condition)
+            condition = self._logic_converter.make_variable(self._logic_converter.convert(condition))
```

Comment:
That line is really a bit strange. Perhaps one of the following two options is a bit nicer: 

1. convert could have a second parameter, something like `define` with is False by default and when it is true, we define the formula with a variable in `DeLogicConverter`.
2. For `DeLogicConverter` the function convert does the `make_variable` and then calls the current `covert` function that gets a different name.

They are both not optimal, but perhaps a bit better to read. What do you think?

Reply:
I think I like option 1 better. Let me try doing that :)